source/state_representation/src/State.cpp

```diff
@@ -43,7 +43,6 @@ void State::set_type(const StateType& type) {
 }
 
 void State::set_name(const std::string& name) {
-  // FIXME: should we prohibit empty names?
   this->name_ = name;
```

Comment:
I would greatly prefer to reset the timestamp here, so it aligns with the design objectives (all non-const methods)

```suggestion
  this->name_ = name;
  this->reset_timestamp();
```